pkg/agent/watcher.go

```diff
@@ -113,7 +120,10 @@ func (w *Watcher) parseConfig(modelConfigs modelconfig.ModelConfigs) {
 			w.modelTracker[name] = modelWrapper{Spec: &spec}
 			w.modelAdded(name, &spec)
 		} else if !cmp.Equal(spec, *existing.Spec) {
-			existing.Spec, existing.stale = &spec, false
+			w.modelTracker[name] = modelWrapper{
+				Spec:  existing.Spec,
+				stale: false,
+			}
 			// Changed - replace
 			w.modelRemoved(name)
 			w.modelAdded(name, &spec)
```

Comment:
Are we keeping this or allowing for updates? I still believe this is v-inefficient